test/unit/test_dwarf.c

```diff
@@ -0,0 +1,720 @@
+#include <r_util.h>
+#include "minunit.h"
+#include <r_bin.h>
+#include <r_core.h>
+#include <r_bin_dwarf.h>
+
+#define MODE 2
+
+// RListComparator should return -1, 0, 1 to indicate "a<b", "a==b", "a>b".
+/**
+ * @brief Comparator to sort list of line statements by address(collection of DwarfRows)
+ */
+int row_comparator(const void *a, const void *b){
+	const RBinDwarfRow *left = a;
+	const RBinDwarfRow *right = b;
+
+	return (left->address >= right->address) ? 1 : -1;
+}
+
+int int_compare(const void *a, const void *b){
+	const int *left = a;
+	const int *right = b;
+	return (*left >= *right) ? 1 : -1;
+}
+
+/**
+ * @brief Tests correct parsing of abbreviations and line information of DWARF3 C binary
+ */
+bool test_dwarf3_c_basic(void) { // this should work for dwarf2 aswell
+	RCore *core = r_core_new ();
+	RBin *bin = r_bin_new ();
+	RIO *io = r_io_new ();
+	r_io_bind (io, &bin->iob);
+
+	RBinOptions opt = { 0 };
+	bool res = r_bin_open (bin, "/home/hound/r2test/dwarf/c/dwarf3", &opt);
+	mu_assert ("couldn't open dwarf3", res);
+
+	// this is probably ugly, but I didn't know how to
+	// tell core  what bin to open so I did it myself
+	r_bin_free (core->bin);
+	core->bin = bin;
```

Comment:
Why do you need core involved at all? Can't you just use `RBin` + `RIO`?

Reply:
Yes, you're right. When I started this testing I was started with looking at bin_dwarf that takes RCore as an argument and didn't look back.